packages/babel-preset-env/src/normalize-options.js

```diff
@@ -15,38 +15,51 @@ const validIncludesAndExcludes = new Set([
   ...defaultWebIncludes,
 ]);
 
-export const validateIncludesAndExcludes = (
-  opts: Array<string> = [],
+const pluginToRegExp = (plugin: any): RegExp => {
+  if (plugin instanceof RegExp) return plugin;
+  try {
+    return new RegExp(`^${normalizePluginName(plugin)}$`);
+  } catch (e) {
+    return null;
+  }
+};
+
+const selectPlugins = (regexp: RegExp): Array<string> =>
+  Array.from(validIncludesAndExcludes).filter(
+    item => regexp instanceof RegExp && item.match(regexp),
+  );
+
+const expandIncludesAndExcludes = (
+  plugins: Array<string | RegExp> = [],
   type: string,
 ): Array<string> => {
-  invariant(
-    Array.isArray(opts),
-    `Invalid Option: The '${type}' option must be an Array<String> of plugins/built-ins`,
-  );
+  if (plugins.length === 0) return plugins;
 
-  const unknownOpts = opts.filter(opt => !validIncludesAndExcludes.has(opt));
+  const selectedPlugins = plugins.map(plugin =>
+    selectPlugins(pluginToRegExp(plugin)),
+  );
+  const invalidRegExpList = plugins.filter(
+    (p, i) => selectedPlugins[i].length === 0,
+  );
 
   invariant(
-    unknownOpts.length === 0,
-    `Invalid Option: The plugins/built-ins '${unknownOpts.join(
+    invalidRegExpList.length === 0,
+    `Invalid Option: The plugins/built-ins '${invalidRegExpList.join(
       ", ",
     )}' passed to the '${type}' option are not
     valid. Please check data/[plugin-features|built-in-features].js in babel-preset-env`,
   );
 
-  return opts;
+  return [].concat(...selectedPlugins);
 };
 
 const validBrowserslistTargets = [
   ...Object.keys(browserslist.data),
   ...Object.keys(browserslist.aliases),
 ];
 
-export const normalizePluginName = (plugin: string): string =>
-  plugin.replace(/^babel-plugin-/, "");
-
-export const normalizePluginNames = (plugins: Array<string>): Array<string> =>
-  plugins.map(normalizePluginName);
+const normalizePluginName = (plugin: string): string =>
+  plugin.replace("babel-plugin-", "");
```

Comment:
why the change? it can replace too much now

Reply:
We already wrap it in `^$` at [line 21](https://github.com/babel/babel/pull/7242/files#diff-1558508a41a0cf543273e188fe7bbf85R21)

Reply:
isnt it wrapped AFTER normalization? with ur approach we can get a final regexp of `/^smth-foo$/` for `smth-babel-plugin-foo` input, which was not the case before the PR, while we are at it maybe you could also add a test case for this?

Reply:
Yes, I made this change to still normalize cases like `"^babel-plugin-foo"`, but I don't think that's a good idea either. I'll add the test and fix this, nice catch 👍